### ISSUE TITLE
Protocol pattern matching non-exhaustive

### ISSUE THREAD

**Issue by layday** [microsoft/pyright#7550]
**Describe the bug**
Runtime-checkable protocols can be used in class patterns. However, when matching an object which is satisfied by a protocol, Pyright still reports that the match statement was non-exhaustive.

**Code or Screenshots**

```python
# pyright: strict

from dataclasses import dataclass
from typing import Protocol, runtime_checkable


@runtime_checkable
class HasX(Protocol):
    __match_args__ = ('x',)
    x: int


@dataclass
class A:
    x: int


match A(1):
    case HasX(x):
        print(f'{x=}')
```

**VS Code extension or command-line**
Either, v1.1.355

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.356, which I just published.
